### DISCUSSION TITLE
Linking Staticly

### DISCUSSION THREAD

**Discussion by BuffaloBingo** [raysan5/raylib#3608] [Q&A] [answered]
Does anybody know of a way to statically link raylib, I am using msys2 and gcc? Any answers would be helpful. I saw that statically linking has been added, but I don't know how to do it.

**Comment by ghost** [comment, marked as answer]
@BuffaloBingo AFAIK statically is the default way. E.g.:
```
cd raylib/src/
make PLATFORM=PLATFORM_DESKTOP
```
The `libraylib.a` will be generated on the src folder. Then, when you compile your program/app, just pass the `-L/path/to/libraylib.a -lraylib` to gcc.

**Reply by BuffaloBingo** [reply]
I got the error. `make: *** empty variable name.  Stop.`

**Reply by ghost** [reply]
Try `mingw32-make PLATFORM=PLATFORM_DESKTOP`.

**Reply by BuffaloBingo** [reply]
I got the error`mingw32-make: *** No targets specified and no makefile found.  Stop.` also I am not using makefile, I am using bash

**Reply by ghost** [reply]
Open your terminal (shell/bash), go to the directory where you unpacked `raylib`, go to the `src` directory there, then try `mingw32-make PLATFORM=PLATFORM_DESKTOP` there.

**Reply by BuffaloBingo** [reply]
I installed via using msys2 and then made a folder inside of home, so I never unpacked raylib. I had msys2 take care of the for me.

**Reply by BuffaloBingo** [reply]
I'm on windows if that helps.

**Reply by ghost** [reply]
`msys2` only provides the tools (`make`, `gcc`, `ar`, etc) necessary to compile code.

If you want to compile `raylib`, you need to download the `raylib` source from this github repository, unpack it, then open your terminal (shell/bash), go to the directory where you unpacked `raylib`, go to the `src` directory there, then `mingw32-make PLATFORM=PLATFORM_DESKTOP` there.

**Reply by ghost** [reply]
@BuffaloBingo As an alternative, you could use the raylib distribution. Here's an excellent video from the Gamefromscratch Youtube channel that shows how to setup it: https://www.youtube.com/watch?v=hr5fYMgjNiE

**Reply by BuffaloBingo** [reply]
what would the GCC command be. I am currently using `g++ "path\to\main.cpp" -o "path\to\out.exe"   $(pkg-config --cflags raylib  --libs  raylib  )`

**Reply by ghost** [reply]
I recommend checking the [raylib wiki](https://github.com/raysan5/raylib/wiki/Working-on-Windows), which has a lot of information about the building process. But the compiling line would usually be something like:
```
g++ /path/to/main.cpp -o /path/to/out.exe -I/path/to/raylib-header-include -L/path/to/raylib-compiled-library -lraylib -lopengl32 -lgdi32 -lwinmm
```

**Reply by orcmid** [reply]
@BuffaloBingo I suggest you take these questions to the raylib Discord server: https://discord.gg/raylib

That's a better place to walk through your current setup, learn what you know around developing on your Window computer, and coming up with a step-by-step approach.  The raylib Discord is equipped for that.  

For example, did you start by installing from the raylib home page, where mingw and the raylib source are installed on your `C:` drive?  (Don't answer here, tell us on Discord.)
